### DISCUSSION TITLE
[i18n] Turn off automatic locale detection

### DISCUSSION THREAD

**Discussion by WonderfulWorlds** [vercel/next.js#18617] [Help]
Is there a way to turn off this feature?

_**Automatic Locale Detection**
When a user visits the application root (generally /), Next.js will try to automatically detect which locale the user prefers based on the Accept-Language header and the current domain.
If a locale other than the default locale is detected, the user will be redirected_

I'd like the site to always display in the default language unless explicitly routed to another language.

Thanks

**Comment by rovaniemi** [comment]
From [canary release docs](https://github.com/vercel/next.js/blob/canary/docs/advanced-features/i18n-routing.md)

The automatic locale detection can be disabled with:

```js
// next.config.js
module.exports = {
  i18n: {
    localeDetection: false,
  },
}
```

When `localeDetection` is set to `false` Next.js will no longer automatically redirect based on the user's preferred locale and will only provide locale information detected from either the locale based domain or locale path as described above.

**Reply by samuelgoff** [reply]
This is broken again, in v`12.1.6`, `12.2.3` & `12.2.4-canary.9`, when you attempt to use `next export`. If you provide the exact config suggested in the [docs](https://nextjs.org/docs/advanced-features/i18n-routing#disabling-automatic-locale-detection) (as above), you get:

```
Error: Specified i18n.locales should be an Array received undefined.
```

If you do what it tells you, more along the lines of what #18482 & other docs suggest, you end up with a config like this:

```
  i18n: {
    locales: ['en-US'],
    defaultLocale: 'en-US',
    localeDetection: false,
  },
```

...and the error changes to this:

```
Error: i18n support is not compatible with next export.
```

Not ideal.
